openweathermap-ios/openweathermap-ios/Weather/Model/WeatherManager.swift

```diff
@@ -0,0 +1,46 @@
+import CoreLocation
+
+protocol WeatherManagerDelegate: class {
+    func didUpdateWeather(weather: WeatherModel)
+}
+
+final class WeatherManager {
+    private let weatherURL = "https://api.openweathermap.org/data/2.5/weather?appid=c6e381d8c7ff98f0fee43775817cf6ad&units=metric"
+    
+    weak var delegate: WeatherManagerDelegate?
+    
+    func fetchWeather(latitude: CLLocationDegrees, longitude: CLLocationDegrees) {
+        let urlString = "\(weatherURL)&lat=\(latitude)&lon=\(longitude)"
+        getRequest(with: urlString)
+    }
+    
+    private func getRequest(with utlString: String) {
+        guard let url = URL(string: utlString) else { return }
+        let session = URLSession(configuration: .default)
+        let task = session.dataTask(with: url) { [unowned self] data, _, erro in
+            if erro != nil {
+                return
+            }
```

Comment:
If you are not using the error, nor giving any feedback to the user, the `guard let data` along with `if let weather` is enough to handle the current logic. 

Consider removing this if statement to keep the code clean.

```suggestion
```